iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/pipe/runtime/PipeMetaSyncer.java

```diff
@@ -104,16 +117,39 @@ private synchronized void sync() {
       pipeAutoRestartRoundCounter.set(0);
     }
 
-    final TSStatus status = procedureManager.pipeMetaSync();
+    final TSStatus metaSyncStatus = procedureManager.pipeMetaSync();
 
     if (somePipesNeedRestarting
-        && status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
-      handleSuccessfulRestartWithLock();
-      procedureManager.pipeHandleMetaChange(true, false);
+        && metaSyncStatus.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      final boolean isRestartSuccessful = handleSuccessfulRestartWithLock();
+      final TSStatus handleMetaChangeStatus =
+          procedureManager.pipeHandleMetaChangeWithBlock(true, false);
+      if (!configManager.getPipeManager().getPipeTaskCoordinator().hasAnyPipe()
+          && isRestartSuccessful
+          && handleMetaChangeStatus.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        LOGGER.info(
+            "set isLastEmptyPipeSyncSuccessful to true, skip next round of sync if no pipe");
+        isLastEmptyPipeSyncSuccessful = true;
```

Comment:
The user behaviour may cause "ABA" problem, such as "empty sync, create, start, drop(partially succeed on some DN)", and the subsequent syncs may be skipped, then the failed DN may never know the drop command and will keep sending data forever.
I think a meta sync cannot be skipped if there are user manual commands after last empty sync.

Reply:
Good catch, I added an extra variable to determine if `PipeTaskInfo` has been modified between two empty sync.